LinkedList/LinkedList/DoublyList.h

```diff
@@ -0,0 +1,429 @@
+#pragma once
+#include <functional>
+
+template <typename Type>
+class TDLList
+{
+private:
+	int Size = 0;
+	struct FNode
+	{
+		Type Data;
+		FNode* Next = nullptr;
+		FNode* Previuos = nullptr;
+	};
+	FNode* Head = nullptr;
+	FNode* Tail = nullptr;
+
+public:
+	TDLList() {}
+
+	~TDLList() 
+	{
+		Clear();
+	}
+
+	//-----------------------------------CopyConstructor
+	TDLList(const TDLList<Type>& OtherList)
+	{
+		Clear();
+		FNode* CurrentNode = OtherList.Head;
+
+		while(CurrentNode)
+		{
+			AddTail(CurrentNode->Data);
+			CurrentNode = CurrentNode->Next;
+		}
+	}
+
+	//------------------------------------Iterator
+	class FIterator
+	{
+	private:
+		FNode* Current;
+	public:
+		FIterator(FNode* NewCurrent)
+		{
+			Current = NewCurrent;
+		}
+
+		FIterator& operator++()
+		{
+			Current = Current->Next;
+			return *this;
+		}
+
+		Type& operator*()
+		{
+			return Current->Data;
+		}
+
+		const Type& operator*() const
+		{
+			return Current->Data;
+		}
+
+		bool operator!=(const FIterator& Other)
+		{
+			return Current != Other.Current;
+		}
+
+		bool operator!=(const FIterator& Other) const
+		{
+			return Current != Other.Current;
+		}
+	};
+
+	FIterator begin()
+	{
+		return FIterator(Head);
+	}
+
+	FIterator end()
+	{
+		return FIterator(nullptr);
+	}
+	
+
+	//---------------------------------Access
+	Type& operator[](const int Index)
+	{
+		FNode* Current;
+		if (Index < Size / 2)
+		{
+			Current = Head;
+			for (int i = 0; i < Index / 2; i++)
```

Comment:
Careful! Why should it go to half of its Index? The half part of the algorithm was already dealt with (starting point if conditional on line 92). This might not bring what you want if you tried to access [4] in an array of 12 elements, for example. It would enter into this for loop, but would only iterate twice (as Index / 2 is 2)